firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -1057,38 +1130,27 @@ class BrowserCoordinator: BaseCoordinator,
 
     // MARK: - Private helpers
 
-    private func showShareSheet(with url: URL?, title: String?) {
-        guard let url else { return }
-
-        let showShareSheet = { url in
-            self.startShareSheetCoordinator(
-                url: url,
-                title: title,
-                sourceView: self.browserViewController.addressToolbarContainer,
-                sourceRect: nil,
-                toastContainer: self.browserViewController.contentContainer,
-                popoverArrowDirection: .any
-            )
-        }
-
-        // We only care about the temp document for .tab shares right?
-        // BUT!! A tab might be displaying a downloaded PDF! Then it's file:// url
-        guard let temporaryDocument = browserViewController.tabManager.selectedTab?.temporaryDocument else {
-            showShareSheet(url)
-            return
-        }
+    private func tryDownloadingTabFileToShare(shareType: ShareType) async -> ShareType {
```

Comment:
This treatment of downloading a file using the TemporaryDocument was not handled equally on all code paths (leading to one bug with FXIOS-10749). Since `startShareSheetCoordinator` calls `tryDownloadingTabFileToShare`, all `.tab` shares now get the same treatment.